python/nav/web/api/v1/views.py

```diff
@@ -1153,3 +1161,45 @@ class ModuleViewSet(NAVAPIMixin, viewsets.ReadOnlyModelViewSet):
         'device__serial',
     )
     serializer_class = serializers.ModuleSerializer
+
+
+class JWTRefreshViewSet(APIView):
+    """
+    Accepts a valid refresh token.
+    Returns a new refresh token and an access token.
+    """
+
+    def post(self, request):
+        incoming_token = request.data.get('refresh_token')
+        token_hash = hash_token(incoming_token)
```

Comment:
I browsed http://127.0.0.1:8000/api/refresh/ and posted an empty request, and this line crashed terribly with an `AttributeError` ('NoneType' object has no attribute 'encode').

Some input validation would be proper here - bad input should result in possibly a *400 Bad Request* (depending on how other parts of the API behave with bad input).

Reply:
Added validation that input was given and that its a string. That should be enough, I dont think its necessary to check that the string has the format of a real token - if its not real then there will be no match for the hash in the db

Reply:
Now it doesn't crash, instead it tells me that my CSRF token (!) is invalid...

Reply:
Added same handling as Vendor endpoint, but it is a bit ugly